presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -170,6 +174,8 @@
     @GuardedBy("this")
     private final SetMultimap<PlanNodeId, ScheduledSplit> pendingSplits = HashMultimap.create();
     @GuardedBy("this")
+    private final Map<PlanNodeId, Long2ObjectMap<ScheduledSplit>> unprocessedSplits = new HashMap<>();
```

Comment:
Correct me if I am wrong but the split sequence id appears to be unique even across different plan nodes. So do you really need this two layer map ?